### DISCUSSION TITLE
F11 and F12

### DISCUSSION THREAD

**Discussion by toml12953** [robhagemans/pcbasic#214] [General]
Both the IBM PC version and PCjr versions of GW-BASIC (BASICA) support 12 function keys but PC-BASIC doesn't even when I choose 
--syntax=pcjr . The Tandy emulator does recognize all 12 function keys. 

Will this be added eventually?

**Comment by robhagemans** [comment]
In what setup did you use GW-BASIC with 12 function keys and what did the
keys do?

The PCjr keyboard itself only had 10 function keys, and GW-BASIC manuals
for the PC also specify only scancodes for 10 function keys. I recall these
keys having no effect in GW-BASIC on DOS machines

**Comment by toml12953** [comment]
The keys had no initial function. I used them for various things in my own programs. Unlike keys 11 and 12 on the Tandy, you didn’t set them by using KEY 11 and 12. You used 30 and 31.

 

KEY 30,”This is key 11”

KEY 31,”This is key 12”

 

Most generic versions of GW-BASIC didn’t use them but IBM’s BASICA and Compaq’s BASICA did.

 

Tom L

**Comment by robhagemans** [comment]
Thanks - do you have any manuals for these or original executables
(assuming those would run on regular MS-DOS)? I'd need some further details
such as scancodes etc. to know what is going on. The manuals I have for IBM
BASICA v1.10 and Compaq BASIC v2 do not mention KEY 30 or function keys
beyond F10.

Also, are you absolutely sure these programs hadn't been updated for use
with qbasic? I do know qbasic had KEY 30 and 31 and of course many programs
were incrementally updated after QBasic came out to use the new features -
making them incompatible with GW-BASIC even though superficially they'd
still look like GW-BASIC programs.

**Comment by toml12953** [comment]
Yes, I’m sure that it was IBM BASICA 3.20 and 3.30 and Compaq GW-BASIC 3.x. I can’t find a PDF online, though.

 

Here’s a scan of some of the pages. Maybe this will help.

 

Tom L

**Comment by robhagemans** [comment]
Thanks for confirming - the scans would be very useful but I don't think
the attachment made it through? At least I can't see it on GitHub. The most
useful bits of the manual would be the description of the KEY statement and
a list of scancodes (usually in an appendix)

**Comment by toml12953** [comment]
I uploaded the BASICA manual pages to the BASIC Programming Language Files area.

 

Tom L

**Comment by robhagemans** [comment]
Hi, apologies, it's still not clear to me where I should look - is that on
a different site? Could you please post a link here?

**Comment by toml12953** [comment]
It’s in the Files area of the Facebook group, BASIC Programming Language. I can’t post it to Github.

**Reply by robhagemans** [reply]
Thanks - I found them now
